haskell/src/Lorentz/Contracts/Stablecoin.hs

```diff
@@ -419,3 +410,101 @@ registryContract =
               $(embedStringFile metadataRegistryContractPath)
         |]
   )
+
+----------------------------------------------------------------------------
+-- TZIP-16 Metadata
+----------------------------------------------------------------------------
+
+metadataMap :: MetadataMap BigMap
+metadataMap = BigMap $ Map.fromList
+  -- One might reasonable expect that the URI would be stored as packed Michelson strings,
+  -- but the TZIP-16 spec is explicit about that not being the case.
+  --
+  -- > Unless otherwise-specified, the encoding of the values must be the direct stream
+  -- > of bytes of the data being stored. (...)
+  -- > There is no implicit conversion to Michelson's binary format (PACK) nor
+  -- > quoting mechanism.
+  --
+  -- See: <https://gitlab.com/tzip/tzip/-/blob/eb1da57684599a266334a73babd7ba82dbbbce66/proposals/tzip-16/tzip-16.md#contract-storage>
+  --
+  -- So, instead, we encode it as UTF-8 byte sequences.
+  [ (mempty, encodeUtf8 @Text "tezos-storage:metadataJSON")
+  , ([mt|metadataJSON|], BSL.toStrict (J.encode metadataJSON))
+  ]
+
+metadataJSON :: Metadata (ToT Storage)
+metadataJSON =
+  Metadata
+    { mName = Just "stablecoin"
+    , mDescription = Nothing
+    , mVersion = Just (toText $ showVersion version)
+    , mLicense = Just License { lName = "MIT", lDetails = Nothing }
+    , mAuthors =
+        [ "Serokell <https://serokell.io/>"
+        , "TQ Tezos <https://tqtezos.com/>"
+        ]
+    , mHomepage = Just "https://github.com/tqtezos/stablecoin/"
+    , mSource = Just Source
+        { sLocation = "https://github.com/tqtezos/stablecoin/tree/v" <> toText (showVersion version) <> "/ligo/stablecoin"
+        , sTools = [ "ligo " ] -- TODO: add ligo version
```

Comment:
> -- TODO: add ligo version

Not sure if left intentional

Reply:
I was thinking we could later parse the ligo compiler version from `nix/sources.json` and insert it here (using TH, for example), but that's not necessary and we can leave that for some other time.